utils/check_bad_commit.py

```diff
@@ -67,20 +77,63 @@ def create_script(target_test):
         fp.write(script.strip())
 
 
+def is_bad_commit(target_test, commit):
+    repo = git.Repo(".")  # or specify path to your repo
+
+    # Save the current HEAD reference
+    original_head = repo.head.commit
+
+    # Checkout to the commit
+    repo.git.checkout(commit)
+
+    create_script(target_test=target_test)
+
+    result = subprocess.run(
+        ["python3", "target_script.py"],
+        capture_output=True,
+        text=True,
+    )
+
+    # Restore to original commit
+    repo.git.checkout(original_head)
+
+    return result.returncode != 0
```

Comment:
used to determine the the commits (start, end) are passing, failing a test or if the test is flaky.

This is important to not give confusing false alert of failing test on a wrong commit.